python/lsst/ts/ess/common/abstract_command_handler.py

```diff
@@ -196,10 +206,97 @@ async def stop_sending_telemetry(self) -> None:
+        self, device_configuration: typing.Dict[str, typing.Any]
+    ) -> BaseDevice:
+        """Get the device to connect to by using the specified configuration.
+
+        Parameters
+        ----------
+        device_configuration: `dict`
+            A dict representing the device to connect to. The format of the
+            dict follows the configuration of the ts_ess_csc project.
+
+        Returns
+        -------
+        device: `common.device.BaseDevice`
+            The device to connect to.
+
+        Raises
+        ------
+        RuntimeError
+            In case an incorrect configuration has been loaded.
+
+        Notes
+        -----
+        In this case a MockDevice always is returned. Sub-classes should
+        override this method to add support for other devices.
+        """
+        sensor = self.get_sensor(device_configuration=device_configuration)
+        self.log.debug(
+            f"Creating MockDevice with name {device_configuration[Key.NAME]} and sensor {sensor}"
+        )
+        device: BaseDevice = MockDevice(
+            name=device_configuration[Key.NAME],
+            device_id=device_configuration[Key.FTDI_ID],
+            sensor=sensor,
+            callback_func=self._callback,
+            log=self.log,
+            disconnected_channel=None,
+        )
+        return device
+
+    def get_sensor(
+        self, device_configuration: typing.Dict[str, typing.Any]
+    ) -> BaseSensor:
+        """Get the sensor to connect to by using the specified configuration.
+
+        Parameters
+        ----------
+        device_configuration: `dict`
+            A dict representing the device to connect to. The format of the
+            dict follows the configuration of the ts_ess_csc project.
+
+        Returns
+        -------
+        sensor: `BaseSensor`
+            The sensor to connect to.
+
+        Raises
+        ------
+        RuntimeError
+            In case an incorrect configuration has been loaded.
+        """
+        if device_configuration[Key.SENSOR_TYPE] == SensorType.HX85A:
+            sensor: BaseSensor = Hx85aSensor(
+                log=self.log,
+            )
+            return sensor
+        elif device_configuration[Key.SENSOR_TYPE] == SensorType.HX85BA:
+            sensor = Hx85baSensor(
+                log=self.log,
+            )
+            return sensor
+        elif device_configuration[Key.SENSOR_TYPE] == SensorType.TEMPERATURE:
+            sensor = TemperatureSensor(
+                log=self.log,
+                num_channels=device_configuration[Key.CHANNELS],
+            )
+            return sensor
+        elif device_configuration[Key.SENSOR_TYPE] == SensorType.WIND:
+            sensor = WindSensor(
+                log=self.log,
+            )
+            return sensor
+        raise RuntimeError(
+            f"Could not get a {device_configuration[Key.SENSOR_TYPE]!r} sensor. "
+            "Please check the configuration."
+        )
```

Comment:
I very strongly suggest that you replace this with a sensor registry: a global dict of SensorType: sensor class.

The current code does not scale well to adding new sensor types and it puts sensor-class-specific knowledge inside of this class, which is unnecessary. Furthermore the long if/else is inefficient and error-prone.

Here's the basic pattern. In a file such as `sensor_registry.py` define:
```
SensorRegistry = dict()

def register_sensor(sensor_type, sensor_class):
    """Add a sensor class (subclass of BaseSensor) to the SensorRegistry.

    ...
    """
    sensor_type = SensorType(sensor_type)
    if not isinstance(sensor_class, BaseSensor):
        raise TypeError(...)
    if sensor_type in SensorRegistry:
        raise ValueError(...)  # duplicate
    SensorRegistry[sensor_type] = sensor_class
```
Then each sensor class you define *registers itself*, which is what makes this scale so well to new sensors. For example in `foo_sensor.py`:
```
from sensor_registry import register_sensor

class FooSensor(BaseSensor):
    ...

register_sensor(SensorType.FOO_SENSOR, FooSensor)
```
Then to create a sensor all you have to do is the following (plus possibly a bit of error handling, if the exceptions that will already be raised aren't clear enough):
```
def make_sensor(device_configuration, log):
    sensor_type = device_configuration[Key.SENSOR_TYPE]
    sensor_class = sensor_registry[sensor_type]
    return sensor_class(log=log)
```